### PR TITLE
adding tests to verify the private registry images

### LINKED ISSUE #2537: Test script requires authentication from registry.redhat.io
[kind/Feature]

## Which functionality do you think we should add?

Actually our test script runs inside a container (https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L10). So for this container, may be I need to get the secret and apply/use that to the running cluster on aws.

To pull the supported image available only in `registry.redhat.io` we need account service credentials to be configured in the main CI and should be mounted to the test container.

## Why is this needed?

To fix the issue where account service credentials are required. eg: Pulling supported builder images from `registry.redhat.io`

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/king test
/kind failing-test

**What does does this PR do / why we need it**:

Verify test for private registry images. As per [comment](https://github.com/openshift/odo/issues/2537#issuecomment-635941209) It should work with multistage

**Which issue(s) this PR fixes**:

Fixes #2537 

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer**:

It should verify tests for private registry successfully as per [comment](https://github.com/openshift/odo/issues/2537#issuecomment-635941209) 

Run `make test-e2e-images`

### CODE REVIEW

File: tests/e2escenarios/e2e_images_test.go [hunk 1]
```diff
@@ -113,4 +113,23 @@ var _ = Describe("odo supported images e2e tests", func() {
 			verifySupportedImage(filepath.Join("rhscl", "nodejs-10-rhel7:latest"), "nodejs", "nodejs:latest", project, appName, context)
 		})
 	})
+
+	Context("odo supported private registry images deployment", func() {
+		JustBeforeEach(func() {
+			if os.Getenv("CI") != "openshift" {
+				Skip("This is an openShift CI specific scenario, skipping")
```
Comment (amitkrout) [inline tests/e2escenarios/e2e_images_test.go:120]: @prietyc123 Registry secrets are by default configured in multistage test infra so it works but this test scenario is not specific to 4.* version. It should work on 3.11 cluster too i mean in travis CI it should run too. 

We have not configured custom secret in travis to access private registry. So please create an issue to track it pass a skip message that makes proper sense. For example
```
       // Issue for configuring login secret for travis CI
       if os.Getenv("CI") != "openshift" {
       Skip("Skipping it on travis CI, skipping")
}

Comment (prietyc123) [reply tests/e2escenarios/e2e_images_test.go]: Tracked the issue https://github.com/openshift/odo/issues/3640 for travis CI.

**Comment by prietyc123** [comment]
```
[odo]  •  Waiting for component to start  ...
 ✓  Waiting for component to start [3s]
[odo] I0724 13:50:24.784500    5616 deployments.go:92] Deployment "wgyrhu" successfully rolled out
[odo] I0724 13:50:24.784515    5616 deployments.go:96] Waiting for deployment spec update to be observed...
[odo] I0724 13:50:24.784569    5616 pods.go:31] Waiting for component=wgyrhu pod
[odo] I0724 13:50:24.881494    5616 pods.go:58] Status of wgyrhu-74f894b7b6-zsr88 pod is Running
[odo] I0724 13:50:24.881519    5616 pods.go:64] Pod wgyrhu-74f894b7b6-zsr88 is Running
[odo] I0724 13:50:24.881586    5616 preference.go:196] The path for preference file is /tmp/873789133/config.yaml
[odo] I0724 13:50:25.149844    5616 url.go:402] Listing ingresses with label selector: app.kubernetes.io/instance=wgyrhu
[odo] I0724 13:50:26.415250    5616 url.go:379] Listing routes with label selector: app.kubernetes.io/part-of=,app.kubernetes.io/instance=wgyrhu
[odo] I0724 13:50:26.415284    5616 occlient.go:2653] Listing routes with label selector: app.kubernetes.io/part-of=,app.kubernetes.io/instance=wgyrhu
[odo]  ✗  Failed To Update Config To Component Deployed.
[odo] Error: unable to list route names: unable to get route list: Get https://127.0.0.1:8443/apis/route.openshift.io/v1/namespaces/kvbrvbyknb/routes?labelSelector=app.kubernetes.io%2Fpart-of%3D%2Capp.kubernetes.io%2Finstance%3Dwgyrhu: unexpected EOF
Deleting project: kvbrvbyknb
Running odo with args [odo project delete kvbrvbyknb -f]
[odo]  ✗  The project kvbrvbyknb does not exist. Please check the list of projects using `odo project list`
• Failure [114.763 seconds]
odo devfile push command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:17
  Handle devfiles with parent
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:685
    should handle a multi layer parent [It]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:711
    No future change is possible.  Bailing out early after 40.184s.
    Running odo with args [odo push --build-command devbuild -f --project kvbrvbyknb]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
```

Restarted the job and the issue already tracked in https://github.com/openshift/odo/issues/3639

**Comment by amitkrout** [comment]
Multistage test infra has by default service account secrets configuration which will authenticate the private registry pull. 

/Approve

**Comment by amitkrout** [comment]
rebased
/lgtm

**Pull request merged**